tests/src/rules/extensions.js

```diff
@@ -624,6 +632,15 @@ describe('TypeScript', () => {
             ],
             parser,
           }),
+          test({
+            code: 'export T from "./typescript-declare";',
```

Comment:
Also, an invalid statement here will throw `Parsing error: Declaration or statement expected`. I am suggesting a named export here as well.

```suggestion
            code: 'export { T } from "./typescript-declare";',
```

Reply:
indeed, https://github.com/tc39/proposal-export-default-from is still a stage 1 proposal